python/ray/data/grouped_dataset.py

```diff
@@ -144,6 +146,72 @@ def _aggregate_on(
         )
         return self.aggregate(*aggs)
 
+    def map_groups(
+        self,
+        fn: Union[CallableClass, Callable[[BatchType], BatchType]],
+        *,
+        compute: Optional[str] = None,
+        batch_format: str = "native",
+        **ray_remote_args
+    ) -> "Dataset[Any]":
+        """Apply a function to each group of records.
```

Comment:
Could you copy over the relevant pydoc from map_batches, matching the doc style?

Reply:
+1, specifically we follow the Google style for Python docstrings: https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

I.e. this docstring should have an `Args:` and `Returns:` section.

Reply:
Done.